### PR TITLE
Rick Dalton hw3 complete

### PR DESCRIPTION
**Pull request by slartib**
Rick Dalton HW3 submission

### CODE REVIEW

File: dataInterface/movies.js [hunk 1]
```diff
@@ -71,6 +80,51 @@ module.exports.getByIdOrTitle = async (identifier) => {
...
+    {
+      throw 'Partial title is required'
+    }
+    const database = client.db(databaseName);
+    const movies = database.collection(collName);
+    console.log(`titleIn: ${titleIn}`)
+    const query = {title: {$regex: titleIn, $options: 'i'}};
+    const fields = {title: 1};
+    let movieCursor = movies.find(query).limit(10).project(fields);
+    return movieCursor.toArray();
```
Comment (cherimarie) [inline dataInterface/movies.js:97]: Fun! Did you get this working, or was it still a work in progress?

File: routes/movies.test.js [hunk 2]
```diff
@@ -31,72 +31,175 @@ describe("/movies routes", () => {
...
+
+      expect(res != null && typeof res === 'object').toEqual(true);
+    });
+    /*
+    it("should return an error message if body is missing title", async () => {
+      movieData.create.mockResolvedValue({error: "Movies must have a title."});
+
+      const res = await request(server).post("/movies/890/comment");
+
+      expect(res.error).toBeDefined();
```
Comment (cherimarie) [inline routes/movies.test.js:128]: ???

**Comment by slartib** [comment]
It was working in the testing I did against database.

Rick

File: routes/movies.test.js [hunk 3]
```diff
@@ -31,72 +31,175 @@ describe("/movies routes", () => {
...
+      const res = await request(server).put("/movies/890");
+      expect(res.error).toBeDefined();
+    });
+  });
+
+  describe("PUT /:movieId/comments/:id", () =>{
+    it("should return the updated comment on success", async () => {
+      movieData.updateCommentById.mockResolvedValue({_id:"890", comment:"Test Comment"});
+      const res = await request(server).put("/movies/890/comments/27");
+      expect(res.body != null && typeof res.body === 'object').toEqual(true);
```
Comment (cherimarie) [inline routes/movies.test.js:156]: While I appreciate the brevity, with unit testing it's valuable to write a separate assertion for each condition we're testing for. Tests often serve as documentation, so it's best that they are very explicit.

File: dataInterface/movies.js [hunk 4]
```diff
@@ -71,6 +80,51 @@ module.exports.getByIdOrTitle = async (identifier) => {
...
+  }
+  finally
+  {
+  }
+}
```
Comment (cherimarie) [inline dataInterface/movies.js:102]: Is the `finally` block necessary?

This is cool- glad to see you stretching!

**Comment by slartib** [comment]
It turned out not to be necessary. I put it in based on some sample code that used it to close the database connection at the end of the call which proved to be a bad idea with the code structure  we were using. I left it in because it didn’t do any harm and it left a structure in place if a future update could benefit from the finally.

**Pull request closed without merging**